crates/loader/src/local.rs

```diff
@@ -525,6 +545,30 @@ fn file_url(path: impl AsRef<Path>) -> Result<String> {
     Ok(Url::from_file_path(abs_path).unwrap().to_string())
 }
 
+fn requires_service_chaining(component: &spin_manifest::schema::v2::Component) -> bool {
+    component
+        .normalized_allowed_outbound_hosts()
+        .unwrap_or_default()
+        .iter()
+        .any(|h| is_chaining_host(h))
+}
+
+fn is_chaining_host(pattern: &str) -> bool {
+    use spin_outbound_networking::{AllowedHostConfig, HostConfig};
+
+    let Ok(allowed) = AllowedHostConfig::parse(pattern) else {
+        return false;
+    };
+
+    match allowed.host() {
+        HostConfig::List(hosts) => hosts
+            .iter()
+            .any(|h| h.ends_with(SERVICE_CHAINING_DOMAIN_SUFFIX)),
+        HostConfig::AnySubdomain(domain) => domain == SERVICE_CHAINING_DOMAIN_SUFFIX,
+        _ => false,
+    }
```

Comment:
I think all of the service chaining host string logic could plausibly live in `spin-outbound-networking`

Reply:
Aaaaaaaaaaggggghhh.  Yes, that was my first thought, but I didn't want to introduce a new dependency for the loader.  But whaddaya know, the loader already depends on `spin-outbound-networking` BUT IT'S NOT LISTED WITH ALL THE OTHER SPIN CRATES.

Ahem.  Yes, I agree.